File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        //using hash map
         HashMap<Integer, Integer> map = new HashMap<>();
         for(int i=0;i<nums.length;i++){
            int diff=target-nums[i];
            if(map.containsKey(diff)){
              return new int[]{map.get(diff), i};
            }

            map.put(nums[i], i);
    }
        

        return new int[]{};
        }
         }








        // int n = nums.length;

        // for (int i = 0; i < n-1; i++) {
            
        //         if (nums[i] + nums[i+1] == target) {
        //             return new int[]{i, i+1};
                
        //     }
        // }

        // return new int[]{};
  

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    //two pointer only works with sorted array
    // let high=nums.length-1;
    // let low=0;
    // let sum=0;
    // while(high>low){
    //     sum=nums[high]+nums[low];
    //     if(sum===target) return [low,high];
    //     if (sum>target) high--;
    //     else low++;

    // }
    // return [];

    let map = new Map();
    for(let i=0;i<nums.length;i++){
        let need= target-nums[i];

        if(map.has(need)) return [i,map.get(need)]

        map.set(nums[i],i)
    }
    
};